staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -287,11 +290,15 @@ func (s *Controller) worker(ctx context.Context) {
 // nodeSyncLoop takes nodeSync signal and triggers nodeSync
 func (s *Controller) nodeSyncLoop(ctx context.Context, workers int) {
 	klog.V(4).Info("nodeSyncLoop Started")
-	for range s.nodeSyncCh {
-		klog.V(4).Info("nodeSync has been triggered")
-		s.nodeSyncInternal(ctx, workers)
+	for {
+		select {
+		case <-s.nodeSyncCh:
+			klog.V(4).Info("nodeSync has been triggered")
+			s.nodeSyncInternal(ctx, workers)
+		case <-ctx.Done():
+			return
+		}
 	}
-	klog.V(2).Info("s.nodeSyncCh is closed. Exiting nodeSyncLoop")
```

Comment:
super nit, but you could keep it,

Reply:
I wanted, but I was getting some klog goroutines hanging forever.
So I got rid of it as it was simpler than debugging klog...

Reply:
Yeah, not worthy